kcidb/db/bigquery/v04_00.py

```diff
@@ -70,7 +71,7 @@ def __init__(self, params):
             project_id = None
             dataset_name = params
         self.client = bigquery.Client(project=project_id)
-        self.dataset_ref = self.client.dataset(dataset_name)
+        self.dataset_ref = DatasetReference(self.client.project, dataset_name)
```

Comment:
What happens if the parameter string lacks the project ID?

Reply:
self.dataset_ref = DatasetReference(dataset_name)
E   TypeError: DatasetReference.__init__() missing 1 required positional argument: 'dataset_id'

Reply:
I was talking about the database parameter string supplied to the KCIDB database client. Read the `_PARAMS_DOC` string above.

With the code currently in main, this command should succeed: `kcidb-db-schema -d bigquery:kcidb_01` (provided you have your `GOOGLE_APPLICATION_CREDENTIALS` set correctly). However, after your change it looks like only this command, specifying the project name explicitly, could work `kcidb-db-schema -d bigquery:kcidb-shivam.kcidb_01`, and the former command would break.

We need to figure out if we could still make the former command to work with the updated API, and if not, update the documentation and all the affected interfaces to reflect that.